src/Beacon.jsx

```diff
@@ -47,31 +48,12 @@ export class Beacon extends Component {
   }
 
   componentWillMount() {
+    const inactive = (this.props.active === false) || (this.context.beacon && (this.context.beacon.active === false));
     const persistent = this.props.persistent || (this.context.beacon && this.context.beacon.persistent);
     const indexedDB = (this.context.beacon && this.context.beacon.indexedDB) || window.indexedDB;
-    this.setState({ persistent, indexedDB });
+    this.setState({ inactive, persistent, indexedDB });
 
-    if (persistent) {
-      // Retrieve the state of the badge from the database
-      // If `persistent` is `true` then we use the hash of the component's children
-      // as a unique ID. `persistent` can be set to some other truthy value to override this default ID.
-      const hash = persistent === true ? sha1(JSON.stringify(this.props.tooltipText)) : persistent;
-      const request = indexedDB.open('react-beacon');
-      request.onupgradeneeded = (event) => {
-        const db = event.target.result;
-        db.createObjectStore('beacons');
-      };
-      request.onsuccess = () => {
-        this.setState({ database: request.result });
-        const transaction = this.state.database.transaction(['beacons']);
-        const objectStore = transaction.objectStore('beacons');
-        objectStore.get(hash).onsuccess = (event) => {
-          if (!event.target.result) {
-            this.setState({ hash });
-          }
-        };
-      };
-    }
+    this.checkHash();
```

Comment:
Moved this to a separate function since we need to do it when we update as well.